bundles/statistics/statsgrid2016/plugin/TogglePlugin.js

```diff
@@ -30,26 +36,49 @@ Oskari.clazz.define('Oskari.statistics.statsgrid.TogglePlugin', function(sandbox
             return this.element;
         }
         var me = this;
-        var toggleButtons = jQuery('<div class="statsgrid-published-toggle-buttons mapplugin"><div class="map"></div><div class="table"></div>');
+        var toggleButtons = jQuery('<div class="statsgrid-published-toggle-buttons mapplugin"><div class="map"></div><div class="table" style="display:none"></div><div class="diagram" style="display:none"></div>');
         var map = toggleButtons.find('.map');
         var table = toggleButtons.find('.table');
+        var diagram = toggleButtons.find('.diagram');
+        var stats = this.sb.findRegisteredModuleInstance('StatsGrid');
```

Comment:
Just pass the reference instead if you need it. But you could just pass the tool "id"s for the UI, make this observable and trigger events so you don't need to have a reference of statsgrid in this class.